GitUI/UserControls/InteractiveGitActionControl.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using Microsoft.VisualStudio.Threading;
+using ResourceManager;
+
+namespace GitUI.UserControls
+{
+    public partial class InteractiveGitActionControl : GitModuleControl
+    {
+        private readonly TranslationString _progressMessage = new TranslationString("{0} is currently in progress.");
+        private readonly TranslationString _progressWithConflictsMessage = new TranslationString("{0} is currently in progress with merge conflicts.");
+
+        private readonly TranslationString _bisect = new TranslationString("Bisect");
+        private readonly TranslationString _rebase = new TranslationString("Rebase");
+        private readonly TranslationString _merge = new TranslationString("Merge");
+        private readonly TranslationString _patch = new TranslationString("Patch");
+
+        public enum State
```

Comment:
These are git operations, not a repo states
Perhaps it should be called 
```suggestion
        public enum GitOperation
```

The rename would flow into the field name and parameter names.

Reply:
I went with GitAction to fit with the name of the control